pkg/operator/apiserver/audit/audit_policies_test.go

```diff
@@ -83,14 +85,58 @@ func TestGetAuditPolicy(t *testing.T) {
 			},
 			goldenFile: "oauth.yaml",
 		},
-		// TODO: add test with multiple customRules
+		{
+			name: "multipleCustomRules",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "system:authenticated:oauth",
+						Profile: "WriteRequestBodies",
+					},
+					{
+						Group:   "system:authenticated",
+						Profile: "AllRequestBodies",
+					},
+				},
+			},
+			goldenFile: "multipleCr.yaml",
+		},
+		{
+			name: "unknownProfile",
+			config: configv1.Audit{
+				Profile: "InvalidString",
+			},
+			errContains: "unknown audit profile \"InvalidString\"",
+		},
+		{
+			name: "unknownCustomRulesProfile",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "InvalidGroup",
+						Profile: "InvalidProfile",
+					},
+				},
+			},
+			errContains: "unknown audit profile \"InvalidProfile\" in customRules for group \"InvalidGroup\"",
+		},
 	}
 	for _, scenario := range scenarios {
 		t.Run(scenario.name, func(t *testing.T) {
 			// act
 			policy, err := GetAuditPolicy(scenario.config)
-			if err != nil {
-				t.Fatal(err)
+			if len(scenario.errContains) == 0 && err != nil {
+				t.Fatalf("Expected no error yet received error: %v", err)
+			}
+			if len(scenario.errContains) > 0 {
+				if err == nil {
+					t.Fatalf("Expected error message: %q, but got: %q", scenario.errContains, err.Error())
```

Comment:
nit: `... : %v", err)`

Reply:
little odd to print err when it is nil. Doesn't that panic? error is an interface. Calling Error on a nil interface panics afaik.

Reply:
This is what Abu recommended here 
https://github.com/openshift/library-go/pull/1126#discussion_r674268845

Reply:
no, my suggestion was not to put it there, `err` is nil here, so it will panic. my suggestion was to put it in the following check where you compare the error messages.

Reply:
The line of code is the same, same conditional. Let's talk more tomorrow.

Reply:
Just went with @sttts  rec here.